types/field_type_test.go

```diff
@@ -171,6 +174,8 @@ func (s *testFieldTypeSuite) TestDefaultTypeForValue(c *C) {
 	}{
 		{nil, mysql.TypeNull, 0, 0, charset.CharsetBin, charset.CharsetBin, mysql.BinaryFlag},
 		{1, mysql.TypeLonglong, 1, 0, charset.CharsetBin, charset.CharsetBin, mysql.BinaryFlag},
+		{0, mysql.TypeLonglong, 1, 0, charset.CharsetBin, charset.CharsetBin, mysql.BinaryFlag},
+		{-42, mysql.TypeLonglong, 2, 0, charset.CharsetBin, charset.CharsetBin, mysql.BinaryFlag},
```

Comment:
This test is wrong, the result should be 3.

Reply:
Aha, I see. I thought it only counts the digits. Thanks for pointing out :)